src/viur/core/prototypes/skelmodule.py

```diff
@@ -108,7 +111,59 @@ def baseSkel(self, *args, **kwargs) -> SkeletonInstance:
 
         By default, baseSkel is used by :func:`~viewSkel`, :func:`~addSkel`, and :func:`~editSkel`.
         """
-        return self._resolveSkelCls(*args, **kwargs)()
+        return self.skel(**kwargs)
+
+    def skel(
+        self,
+        *,
+        allow_client_defined: bool = False,
+        bones: tuple[str, ...] | t.List[str] = (),
+        exclude_bones: tuple[str, ...] | t.List[str] = (),
+        **kwargs,
+    ) -> SkeletonInstance:
+        """
+        Retrieve module-specific skeleton, optionally as subskel.
+
+        :param allow_client_defined: Evaluates header X-VIUR-BONELIST to contain a comma-separated list of bones.
+            Using this parameter enforces that the Skeleton class has a subskel named "*" for required bones that
+            must exist.
+        :param bones: Allows to specify a list of bones to form a subskel.
+        :param exclude_bones: Provide a list of bones which are always excluded.
+
+        The parameters `bones` and `allow_client_defined` can be combined.
+        """
+        skel_cls = self._resolveSkelCls(**kwargs)
+        bones = set(bones) if bones else set()
+
+        if allow_client_defined:
+            # if bonelist := current.request.get().kwargs.get(X_VIUR_BONELIST.lower()):  # DEBUG
```

Comment:
@sveneberth I wonder if this could also be useful for debugging, what do you think?

Reply:
I don't need that. Sending the header works perfectly for me, even when testing.